js/ui/paygUnlockDialog.js

```diff
@@ -342,7 +342,14 @@ var PaygUnlockDialog = new Lang.Class({
         } else if (error.matches(PaygManager.PaygErrorDomain, PaygManager.PaygError.CODE_ALREADY_USED)) {
             this._setErrorMessage(_("Code already used. Please enter a new code."));
         } else if (error.matches(PaygManager.PaygErrorDomain, PaygManager.PaygError.TOO_MANY_ATTEMPTS)) {
-            this._setErrorMessage(_("Too many attempts. Please wait and try again."));
+            let currentTime = GLib.get_real_time() / GLib.USEC_PER_SEC;
+            let secondsLeft = Main.paygManager.rateLimitEndTime - currentTime;
+            let minutesLeft = Math.max(0, Math.round(secondsLeft / 60))
+            if (minutesLeft > 0)
+                this._setErrorMessage(_("Too many attempts. Try again in %s %s.")
+                                      .format(minutesLeft, (minutesLeft > 1) ? _("minutes") : _("minute")));
```

Comment:
Is there no `ngettext()` in JS? This method of doing plurals will not work for many languages.

What about if `Main.paygManager.rateLimitEndTime < currentTime`? `secondsLeft` will be negative then. I guess the `(minutesLeft > 0)` condition will fail then, but that seems a bit unsatisfying.

Reply:
It's available as `Gettext.ngettext()`

Reply:
> Is there no ngettext() in JS? This method of doing plurals will not work for many languages.

Oops! My bad... I didn't remember the way to do it and I put this as a placeholder, then I forgot. TL;DR: Yes, we can use ngettext(), I'll change that.

> What about if Main.paygManager.rateLimitEndTime < currentTime? secondsLeft will be negative then. I guess the (minutesLeft > 0) condition will fail then, but that seems a bit unsatisfying.

That should be handled by `Math.max(0, Math.round(secondsLeft / 60))` no?

Reply:
I'll change the condition to `minutesLeft >= 0` to make it more clear that the `else` branch handles the `[0, 1)` interval